### PR TITLE
COMPAT: followup to #17491

### PR DESCRIPTION
**Pull request by jreback**

### CODE REVIEW

File: doc/source/whatsnew/v0.21.0.txt [hunk 1]
```diff
@@ -193,16 +193,16 @@ the target. Now, a ``ValueError`` will be raised when such an input is passed in
 Iteration of Series/Index will now return python scalars
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Previously, when using certain iteration methods for a ``Series`` with dtype ``int`` or ``float``, you would receive a ``numpy`` scalar, e.g. a ``np.int64``, rather than a python ``int``. Issue (:issue:`10904`) corrected this for ``Series.tolist()`` and ``list(Series)``. This change makes all iteration methods consistent, in particular, for ``__iter__()`` and ``.map()``; note that this only affect int/float dtypes. (:issue:`13236`, :issue:`13258`, :issue:`14216`).
+Previously, when using certain iteration methods for a ``Series`` with dtype ``int`` or ``float``, you would receive a ``numpy`` scalar, e.g. a ``np.int64``, rather than a python ``int``. Issue (:issue:`10904`) corrected this for ``Series.tolist()`` and ``list(Series)``. This change makes all iteration methods consistent, in particular, for ``__iter__()`` and ``.map()``; note that this only affects int/float dtypes. (:issue:`13236`, :issue:`13258`, :issue:`14216`).
```
Comment (gfyoung) [inline doc/source/whatsnew/v0.21.0.txt]: python --> Python

File: pandas/core/base.py [hunk 2]
```diff
@@ -892,18 +892,27 @@ def argmin(self, axis=None):
 
     def tolist(self):
         """
-        return a list of the values; box to scalars
+        return a list of the values. These are each a scalar type, which is
```
Comment (gfyoung) [inline pandas/core/base.py]: Capitalize that first sentence.

File: pandas/tests/test_base.py [hunk 3]
```diff
@@ -1102,3 +1136,40 @@ def test_categorial_datetimelike(self, method):
 
         result = method(i)[0]
         assert isinstance(result, Timestamp)
+
+    def test_iter_box(self):
```
Comment (gfyoung) [inline pandas/tests/test_base.py]: Is this just a new test for coverage purposes or related to any of the issues you addressed?

Comment (jreback) [reply pandas/tests/test_base.py]: a moved test

**Review by gfyoung — APPROVED** [review]

File: pandas/core/indexes/category.py [hunk 4]
```diff
@@ -253,9 +253,8 @@ def get_values(self):
         """ return the underlying data as an ndarray """
         return self._data.get_values()
 
-    def __iter__(self):
-        """ iterate like Categorical """
-        return self._data.__iter__()
+    def tolist(self):
+        return self._data.tolist()
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/category.py]: I don't think this is needed? Base `Index` already defines this as `self._values.tolist()`, and `_values` and `_data` is both a Categorical ?

**Pull request merged**